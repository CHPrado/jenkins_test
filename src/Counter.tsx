import React from 'react';

interface CounterProps {
    count: number;
}

const Counter = (props: CounterProps) => {
  return (
    <div className="Counter">
      <h2>Você tem { props.count } counters;</h2>
    </div>
  );
}

export default Counter;
