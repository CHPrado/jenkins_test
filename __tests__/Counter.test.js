import React from "react";
import { mount } from 'enzyme';
import Counter from "../src/Counter";

describe('Testing Counter Component', () => {
  it('should render correctly', () => {
    const wrapper = mount(
      <Counter count={5} />
    );

    expect(wrapper.debug()).toMatchSnapshot();
    wrapper.setProps({ count: 10*2 });
    expect(wrapper.debug()).toMatchSnapshot();
  });
});