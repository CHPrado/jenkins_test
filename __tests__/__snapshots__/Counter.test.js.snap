// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Counter Component should render correctly 1`] = `
"<Counter count={5}>
  <div className=\\"Counter\\">
    <h2>
      Voc� tem 
      5
       counters;
    </h2>
  </div>
</Counter>"
`;

exports[`Testing Counter Component should render correctly 2`] = `
"<Counter count={20}>
  <div className=\\"Counter\\">
    <h2>
      Voc� tem 
      20
       counters;
    </h2>
  </div>
</Counter>"
`;
